lms/services/digest.py

```diff
@@ -73,11 +73,17 @@ def send_instructor_email_digests(  # pylint:disable=too-many-arguments
 
             if deduplicate:
                 task_done_key = f"instructor_email_digest::{user_info.h_userid}::{datetime.now(timezone.utc).strftime('%Y-%m-%d')}"
+                task_done_data = {
+                    "type": "instructor_email_digest",
+                    "created_before": created_before.isoformat(),
+                }
             else:
                 task_done_key = None
+                task_done_data = None
```

Comment:
If `task_done_key` is `None` then `MailchimpService` doesn't even add a row to the `task_done` table (since you can't add a row without a `key`) so there's no point in passing it any `task_done_data` in this case either.

A better design here might be for `DigestService` itself to actually create the `TaskDone` object rather than letting `MailchimpService` create it, then you'd have a single `task_done` argument to `MailchimpService` rather than separate `task_done_key` and `task_done_data` arguments that must always go together. But we're not calling `MailchimpService` directly, we're calling it via Celery, so that would require us to make `TaskDone`'s serializable.